### ISSUE TITLE
[Firefox] cannot download latest update of umatrix

### ISSUE THREAD

**Issue by excalibur1234** [gorhill/uMatrix#209]
i am using firefox developer edition and the latest .dev build of umatrix.
also, i have unchecked ALL privacy settings and whitelist EVERYTHING on https://github.com/gorhill/uMatrix/releases.

still, i am not able to download the latest .xpi file from https://github.com/gorhill/uMatrix/releases.
firefox cannot update umatrix automatically.

when i disable the umatrix addon, i can download the latest .xpi file from https://github.com/gorhill/uMatrix/releases

can anybody reproduce this problem?

**Comment by gorhill** [comment]
A primary tool for diagnosing issue is to use the logger. What does it say when you try to download?

**Comment by excalibur1234** [comment]
sorry. i forgot about that.
here is the log: http://pastebin.com/9qD8yJxK

**Comment by gorhill** [comment]
You've got your answer in there, unblock the `other` cell.

**Comment by excalibur1234** [comment]
thanks a lot! it works now.

for everybody having the same problem, here a short guide how to solve it. this is not an intuitive solution and i would never have figured it out by myself
- go to https://github.com/gorhill/uMatrix/releases and start to download a .xpi add-on file for firefox.
- completely whitelisting the page is useless. you will not be able to download the .xpi file: http://imagizer.imageshack.com/img633/6849/53XWOQ.png
- solution: you need to whitelist at least the "other" in "s3.amazonaws.com" globally, despite umatrix not showing any blocked content: http://imagizer.imageshack.com/img910/8272/eQIMvO.png

**Comment by gorhill** [comment]
> despite umatrix not showing any blocked content

Was it because this was a behind-the-scene request?

**Comment by requiredregistration** [comment]
he must be using a download manager extension, because without download manager extensions it works fine.

**Comment by excalibur1234** [comment]
your solution probably works with firefox 39 and lower (i have not tested this).
i am using firefox 40. it does not work, because it wants to verify the download first.
i think it is has to do something with these new features of firefox 40:
https://wiki.mozilla.org/Addons/Extension_Signing
https://wiki.mozilla.org/Electrolysis

my solution works for firefox 40, too. it is not perfect at all, but it works at the moment:
* s3.amazonaws.com other allow

but this problem is probably way off topic. open another issue, if you think so, too.
